packages/fluentui/react-northstar/src/components/Form/FormFieldCustom.tsx

```diff
@@ -0,0 +1,215 @@
+
+  /** A label for the form field. */
+  label?: ShorthandValue<TextProps>;
+
+  /** Text message that will be displayed below the control (can be used for error, warning, success messages). */
+  message?: ShorthandValue<TextProps>;
+
+  /** Message to be shown when input has error */
+  errorMessage?: ShorthandValue<TextProps>;
+}
+
+export const formFieldClassName = 'ui-form__field';
+export const formFieldMessageClassName = 'ui-form__field__message';
+
+export type FormFieldCustomStylesProps = Required<Pick<FormFieldCustomProps, 'inline'>> & {
+  hasErrorMessage: boolean;
+};
+
+const FormFieldCustom = compose<'div', FormFieldCustomProps, FormFieldCustomStylesProps, {}, {}>(
+  (props, ref, composeOptions) => {
+    const context: ProviderContextPrepared = React.useContext(ThemeContext);
+    const { setStart, setEnd } = useTelemetry(composeOptions.displayName, context.telemetry);
+    setStart();
+
+    const { children, message, className, design, styles, variables, inline, errorMessage, id, control, label } = props;
+
+    const slotProps = composeOptions.resolveSlotProps<FormFieldCustomProps>(props);
+    const ElementType = getElementType(props);
+    const unhandledProps = useUnhandledProps(composeOptions.handledProps, props);
+    const messageId = React.useRef<string>();
+    messageId.current = getOrGenerateIdFromShorthand('error-message-', message || errorMessage, messageId.current);
+    const labelId = React.useRef<string>();
+    labelId.current = getOrGenerateIdFromShorthand('form-label-', id, labelId.current);
+
+    const getA11yProps = useAccessibility<FormFieldBehaviorProps>(props.accessibility, {
+      debugName: composeOptions.displayName,
+      mapPropsToBehavior: () => ({
+        hasErrorMessage: !!errorMessage,
+        messageId: messageId.current,
+        labelId: labelId.current,
+      }),
+      rtl: context.rtl,
+    });
+
+    const { classes, styles: resolvedStyles } = useStyles<FormFieldCustomStylesProps>(FormFieldCustom.displayName, {
+      className: composeOptions.className,
+      composeOptions,
+      mapPropsToStyles: () => ({
+        inline,
+        hasErrorMessage: !!errorMessage,
+      }),
+      mapPropsToInlineStyles: () => ({
+        className,
+        design,
+        styles,
+        variables,
+      }),
+      rtl: context.rtl,
+      unstable_props: props,
+    });
+
+    const labelElement = createShorthand(composeOptions.slots.label, label, {
+      defaultProps: () =>
+        getA11yProps('label', {
+          as: 'label',
+          htmlFor: id,
+          id: labelId.current,
+          styles: resolvedStyles.label,
+          inline,
+          ...slotProps.label,
+        }),
+    });
+
+    const messageElement = createShorthand(composeOptions.slots.message, errorMessage || message, {
+      defaultProps: () =>
+        getA11yProps('message', {
+          className: formFieldMessageClassName,
+          id: messageId.current,
+          styles: resolvedStyles.message,
```

Comment:
```suggestion
```

Reply:
we don't have `formMessageStyles` it's resolved in the `formFieldStyles`

Reply:
Done